mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -320,6 +362,23 @@ class KMeans @Since("1.5.0") (
       .setMaxIterations($(maxIter))
       .setSeed($(seed))
       .setEpsilon($(tol))
+
+    if (isDefined(initialModel)) {
+      // Check the equal of dimension
+      val dimOfData = rdd.first().size
+      val dimOfInitialModel = $(initialModel).clusterCenters.head.size
+      require(dimOfData == dimOfInitialModel,
+        s"mismatched dimension, $dimOfData in data while $dimOfInitialModel in the initial model.")
+
+      // Check the equal of number of clusters
```

Comment:
very minor, but this doesn't make sense. Maybe `// Check that the number of clusters are equal`